src/test/java/spoon/test/fieldaccesses/FieldAccessTest.java

```diff
@@ -223,4 +227,27 @@ public void testIncrementationOnAVarIsAUnaryOperator() throws Exception {
 		assertEquals(fieldRead, second.getOperand());
 		assertEquals("++(i)", second.toString());
 	}
+
+	@Test
+	public void testFieldWriteWithPlusEqualsOperation() throws Exception {
+		// contract: When we use var += value, the var is a write access.
+		final CtType<Panini> aPanini = TestUtils.buildClass(Panini.class);
+		final CtMethod<?> prepare = aPanini.getMethodsByName("prepare").get(0);
+
+		final List<CtFieldWrite<?>> fields = prepare.getElements(new TypeFilter<>(CtFieldWrite.class));
+		assertEquals(1, fields.size());
+		assertEquals(aPanini.getField("i").getReference(), fields.get(0).getVariable());
+		assertEquals("i += 0", fields.get(0).getParent().toString());
+		assertEquals("i", fields.get(0).toString());
+
+		final List<CtVariableWrite<?>> variables = prepare.getElements(new TypeFilter<>(CtVariableWrite.class));
+		assertEquals(1, variables.size());
+		assertEquals("j += 0", variables.get(0).getParent().toString());
+		assertEquals("j", variables.get(0).toString());
+
+		final List<CtArrayWrite<?>> arrays = prepare.getElements(new TypeFilter<>(CtArrayWrite.class));
+		assertEquals(1, arrays.size());
+		assertEquals("array[0] += 0", arrays.get(0).getParent().toString());
+		assertEquals("array[0]", arrays.get(0).toString());
```

Comment:
Why do you never check the type of ```arrays.get(0)```?
Here you test the DefaultJavaPrettyPrinter and not the model.